<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CaveWindow</class>
 <widget class="QDialog" name="CaveWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>950</width>
    <height>650</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>950</width>
    <height>650</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>950</width>
    <height>650</height>
   </size>
  </property>
  <property name="baseSize">
   <size>
    <width>950</width>
    <height>650</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Cave</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QWidget {
	background-color: #0C090A;
	color: rgb(255, 255, 255);
}</string>
  </property>
  <widget class="CaveScene" name="caveScene" native="true">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>50</y>
     <width>500</width>
     <height>500</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
   <property name="baseSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
  </widget>
  <widget class="QPushButton" name="loadButtonCave">
   <property name="geometry">
    <rect>
     <x>670</x>
     <y>70</y>
     <width>261</width>
     <height>40</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>JetBrains Mono</family>
     <pointsize>-1</pointsize>
     <bold>false</bold>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QPushButton {
	background-color: qlineargradient(spread: pad, x1: 0, y1: 0, x2: 1, y2: 1,
											  stop:0 #CC0099, stop:1 #FF9933);
	color: #FFFFFF;
	border-style: solid;
	border-color: #FFFFFF;
	border-width: 0px;
	font-size: 20px;
}

QPushButton::hover {
	border-width: 4px;
}

QPushButton::pressed {
	border-width: 0px;
	background-color:  #0C090A;
	color:  qlineargradient(spread: pad, x1: 0, y1: 0, x2: 1, y2: 1,
											  stop:0 #CC0099, stop:1 #FF9933);
}</string>
   </property>
   <property name="text">
    <string>LOAD</string>
   </property>
  </widget>
  <widget class="QLabel" name="fileNameLabel">
   <property name="geometry">
    <rect>
     <x>670</x>
     <y>10</y>
     <width>261</width>
     <height>50</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>JetBrains Mono</family>
     <pointsize>-1</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QLabel {
	color: #FFFFFF;
	font-size: 20px;
}</string>
   </property>
   <property name="text">
    <string>FILENAME</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QPushButton" name="generateButton">
   <property name="geometry">
    <rect>
     <x>670</x>
     <y>440</y>
     <width>260</width>
     <height>40</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>JetBrains Mono</family>
     <pointsize>-1</pointsize>
     <bold>false</bold>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QPushButton {
	background-color: qlineargradient(spread: pad, x1: 0, y1: 0, x2: 1, y2: 1,
											  stop:0 #CC0099, stop:1 #FF9933);
	color: #FFFFFF;
	border-style: solid;
	border-color: #FFFFFF;
	border-width: 0px;
	font-size: 20px;
}

QPushButton::hover {
	border-width: 4px;
}

QPushButton::pressed {
	border-width: 0px;
	background-color:  #0C090A;
	color:  qlineargradient(spread: pad, x1: 0, y1: 0, x2: 1, y2: 1,
											  stop:0 #CC0099, stop:1 #FF9933);
}</string>
   </property>
   <property name="text">
    <string>GENERATE</string>
   </property>
  </widget>
  <widget class="QLabel" name="widthLabel">
   <property name="geometry">
    <rect>
     <x>670</x>
     <y>180</y>
     <width>120</width>
     <height>40</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>JetBrains Mono</family>
     <pointsize>-1</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QLabel {
	color: #FFFFFF;
	font-size: 20px;
}</string>
   </property>
   <property name="text">
    <string>WIDTH</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QSpinBox" name="rowsSpinBox">
   <property name="geometry">
    <rect>
     <x>800</x>
     <y>180</y>
     <width>110</width>
     <height>40</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>JetBrains Mono</family>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QSpinBox {
	color: #FFFFFF;
}</string>
   </property>
   <property name="minimum">
    <number>1</number>
   </property>
   <property name="maximum">
    <number>50</number>
   </property>
   <property name="value">
    <number>20</number>
   </property>
  </widget>
  <widget class="QLabel" name="heightLabel">
   <property name="geometry">
    <rect>
     <x>670</x>
     <y>230</y>
     <width>120</width>
     <height>40</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>JetBrains Mono</family>
     <pointsize>-1</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QLabel {
	color: #FFFFFF;
	font-size: 20px;
}</string>
   </property>
   <property name="text">
    <string>HEIGHT</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QSpinBox" name="colsSpinBox">
   <property name="geometry">
    <rect>
     <x>800</x>
     <y>230</y>
     <width>110</width>
     <height>40</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>JetBrains Mono</family>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QSpinBox {
	color: #FFFFFF;
}</string>
   </property>
   <property name="minimum">
    <number>1</number>
   </property>
   <property name="maximum">
    <number>50</number>
   </property>
   <property name="value">
    <number>20</number>
   </property>
  </widget>
  <widget class="QPushButton" name="stepButton">
   <property name="geometry">
    <rect>
     <x>670</x>
     <y>490</y>
     <width>261</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>JetBrains Mono</family>
     <pointsize>-1</pointsize>
     <bold>false</bold>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QPushButton {
	background-color: qlineargradient(spread: pad, x1: 0, y1: 0, x2: 1, y2: 1,
											  stop:0 #CC0099, stop:1 #FF9933);
	color: #FFFFFF;
	border-style: solid;
	border-color: #FFFFFF;
	border-width: 0px;
	font-size: 20px;
}

QPushButton::hover {
	border-width: 4px;
}

QPushButton::pressed {
	border-width: 0px;
	background-color:  #0C090A;
	color:  qlineargradient(spread: pad, x1: 0, y1: 0, x2: 1, y2: 1,
											  stop:0 #CC0099, stop:1 #FF9933);
}</string>
   </property>
   <property name="text">
    <string>NEXT STEP</string>
   </property>
  </widget>
  <widget class="QPushButton" name="autoButton">
   <property name="geometry">
    <rect>
     <x>670</x>
     <y>600</y>
     <width>261</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>JetBrains Mono</family>
     <pointsize>-1</pointsize>
     <bold>false</bold>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QPushButton {
	background-color: qlineargradient(spread: pad, x1: 0, y1: 0, x2: 1, y2: 1,
											  stop:0 #CC0099, stop:1 #FF9933);
	color: #FFFFFF;
	border-style: solid;
	border-color: #FFFFFF;
	border-width: 0px;
	font-size: 20px;
}

QPushButton::hover {
	border-width: 4px;
}

QPushButton::pressed {
	border-width: 0px;
	background-color:  #0C090A;
	color:  qlineargradient(spread: pad, x1: 0, y1: 0, x2: 1, y2: 1,
											  stop:0 #CC0099, stop:1 #FF9933);
}</string>
   </property>
   <property name="text">
    <string>AUTO</string>
   </property>
  </widget>
  <widget class="QLabel" name="frequencyLabel">
   <property name="geometry">
    <rect>
     <x>670</x>
     <y>550</y>
     <width>121</width>
     <height>40</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>JetBrains Mono</family>
     <pointsize>-1</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QLabel {
	color: #FFFFFF;
	font-size: 16px;
}</string>
   </property>
   <property name="text">
    <string>FREQUENCY(MS)</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
   </property>
  </widget>
  <widget class="QSpinBox" name="frequencySpinBox">
   <property name="geometry">
    <rect>
     <x>799</x>
     <y>550</y>
     <width>111</width>
     <height>40</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>JetBrains Mono</family>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QSpinBox {
	color: #FFFFFF;
}</string>
   </property>
   <property name="minimum">
    <number>300</number>
   </property>
   <property name="maximum">
    <number>2000</number>
   </property>
   <property name="value">
    <number>500</number>
   </property>
  </widget>
  <widget class="QPushButton" name="saveButtonCave">
   <property name="geometry">
    <rect>
     <x>670</x>
     <y>120</y>
     <width>260</width>
     <height>40</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>JetBrains Mono</family>
     <pointsize>-1</pointsize>
     <bold>false</bold>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QPushButton {
	background-color: qlineargradient(spread: pad, x1: 0, y1: 0, x2: 1, y2: 1,
											  stop:0 #CC0099, stop:1 #FF9933);
	color: #FFFFFF;
	border-style: solid;
	border-color: #FFFFFF;
	border-width: 0px;
	font-size: 20px;
}

QPushButton::hover {
	border-width: 4px;
}

QPushButton::pressed {
	border-width: 0px;
	background-color:  #0C090A;
	color:  qlineargradient(spread: pad, x1: 0, y1: 0, x2: 1, y2: 1,
											  stop:0 #CC0099, stop:1 #FF9933);
}</string>
   </property>
   <property name="text">
    <string>SAVE</string>
   </property>
  </widget>
  <widget class="QLabel" name="deathLabel">
   <property name="geometry">
    <rect>
     <x>670</x>
     <y>280</y>
     <width>120</width>
     <height>40</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>JetBrains Mono</family>
     <pointsize>-1</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QLabel {
	color: #FFFFFF;
	font-size: 16px;
}</string>
   </property>
   <property name="text">
    <string>DEATH LIMIT</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QSpinBox" name="deathSpinBox">
   <property name="geometry">
    <rect>
     <x>800</x>
     <y>280</y>
     <width>110</width>
     <height>40</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>JetBrains Mono</family>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QSpinBox {
	color: #FFFFFF;
}</string>
   </property>
   <property name="minimum">
    <number>0</number>
   </property>
   <property name="maximum">
    <number>7</number>
   </property>
  </widget>
  <widget class="QLabel" name="birthLabel">
   <property name="geometry">
    <rect>
     <x>670</x>
     <y>330</y>
     <width>120</width>
     <height>40</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>JetBrains Mono</family>
     <pointsize>-1</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QLabel {
	color: #FFFFFF;
	font-size: 16px;
}</string>
   </property>
   <property name="text">
    <string>BIRTH LIMIT</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QSpinBox" name="birthSpinBox">
   <property name="geometry">
    <rect>
     <x>800</x>
     <y>330</y>
     <width>110</width>
     <height>40</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>JetBrains Mono</family>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QSpinBox {
	color: #FFFFFF;
}</string>
   </property>
   <property name="minimum">
    <number>0</number>
   </property>
   <property name="maximum">
    <number>7</number>
   </property>
  </widget>
  <widget class="QLabel" name="chanceLabel">
   <property name="geometry">
    <rect>
     <x>670</x>
     <y>380</y>
     <width>120</width>
     <height>40</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>JetBrains Mono</family>
     <pointsize>-1</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QLabel {
	color: #FFFFFF;
	font-size: 16px;
}</string>
   </property>
   <property name="text">
    <string>BIRTH CHANCE</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QSpinBox" name="chanceSpinBox">
   <property name="geometry">
    <rect>
     <x>800</x>
     <y>380</y>
     <width>110</width>
     <height>40</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>JetBrains Mono</family>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QSpinBox {
	color: #FFFFFF;
}</string>
   </property>
   <property name="minimum">
    <number>1</number>
   </property>
   <property name="maximum">
    <number>100</number>
   </property>
   <property name="value">
    <number>50</number>
   </property>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CaveScene</class>
   <extends>QWidget</extends>
   <header>view/cave_scene.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
